engines/native/sandbox.go

```diff
@@ -47,11 +47,20 @@ func newSandbox(b *sandboxBuilder) (*sandbox, error) {
 		}
 	}
 
-	// Create temporary user account
-	user, err := system.CreateUser(homeFolder.Path(), b.engine.groups)
-	if err != nil {
-		homeFolder.Remove() // best-effort clean-up this is a fatal error
-		return nil, fmt.Errorf("Failed to create temporary system user, error: %s", err)
+	var user *system.User
+
+	if b.engine.config.CreateUser {
+		// Create temporary user account
+		user, err = system.CreateUser(homeFolder.Path(), b.engine.groups)
+		if err != nil {
+			homeFolder.Remove() // best-effort clean-up this is a fatal error
+			return nil, fmt.Errorf("Failed to create temporary system user, error: %s", err)
+		}
+	} else {
+		user, err = system.CurrentUser()
+		if err != nil {
+			return nil, err
+		}
```

Comment:
Maybe we should assign `nil` to `user` as a flag to say "we are running as current user", so, in case someone mistakenly tries to delete the current user, we get a `panic`, instead of a dead host.